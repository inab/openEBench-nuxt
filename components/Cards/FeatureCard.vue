<template>
	<div>
		<v-avatar color="primary" size="62" class="mb-5 mb-md-10">
			<slot name="icon" />
		</v-avatar>
		<h2 class="text-h6 mb-2"><slot name="title" /></h2>
		<p class="body-1">
			<slot name="description" />
		</p>
		<v-btn class="mt-2" outlined color="primary" :to="to"> Explore </v-btn>
	</div>
</template>

<script>
export default {
	name: 'FeatureCard',
	props: {
		to: {
			type: String,
			required: true,
		},
	},
};
</script>
