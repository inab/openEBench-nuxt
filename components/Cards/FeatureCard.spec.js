import { shallowMount } from '@vue/test-utils';
import FeatureCard from './FeatureCard.vue';

const factory = () => {
	return shallowMount(FeatureCard, {
		...createComponentMocks({}),
		mocks: {
			$vuetify: { breakpoint: {} },
		},
		propsData: { to: 'someroute' },
	});
};

describe('FeatureCard', () => {
	it('is instantiated', () => {
		const wrapper = factory();
		expect(wrapper).toBeTruthy();
	});

	it('should match snapshot', () => {
		const wrapper = factory();
		expect(wrapper.html()).toMatchSnapshot();
	});
});
