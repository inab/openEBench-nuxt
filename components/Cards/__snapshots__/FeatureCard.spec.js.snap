// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeatureCard should match snapshot 1`] = `
"<div>
  <v-avatar-stub color=\\"primary\\" size=\\"62\\" class=\\"mb-5 mb-md-10\\"></v-avatar-stub>
  <h2 class=\\"text-h6 mb-2\\"></h2>
  <p class=\\"body-1\\"></p>
  <v-btn-stub color=\\"primary\\" outlined=\\"true\\" tag=\\"button\\" activeclass=\\"\\" to=\\"someroute\\" type=\\"button\\" class=\\"mt-2\\"> Explore </v-btn-stub>
</div>"
`;
